std/experimental/testing/testsuite.d

```diff
@@ -0,0 +1,298 @@
+module std.experimental.testing.testsuite;
+
+import std.experimental.testing.testcase;
+import std.experimental.testing.io;
+import std.experimental.testing.options;
+import std.datetime;
+import std.parallelism: taskPool;
+import std.algorithm;
+import std.conv: text;
+import std.array;
+import core.runtime;
+
+
+
+/**
+ * Only exists because taskPool.amap only works with regular functions,
+ * not closures.
+ */
+auto runTest(TestCase test)
+{
+    return test();
+}
+
+/**
+ * Responsible for running tests and printing output.
+ */
+struct TestSuite
+{
+    /**
+    Params:
+      options = The options to run tests with.
+      testData = The information about the tests to run.
+    */
+    this(in Options options, in TestData[] testData)
```

Comment:
Ah. I still think it's unnecessary and premature optmisation. I wish I could find Andrei's quote in the forums. It was all about rvalues and const ref.